### PR TITLE
Add upgrade bats tests

### PR DESCRIPTION
**Pull request by ehelms**
This is what I used to test 3.1 to 3.3 upgrades @jlsherrill

### CODE REVIEW

File: bats/fb-content-katello.bats [hunk 1]
```diff
@@ -141,7 +142,7 @@ EOF
 }
 
 @test "enable content view repo" {
-  subscription-manager repos --enable="Test_Organization_Test_Product_Zoo" | grep -q "is enabled for this system"
+  subscription-manager repos --enable="Test_Organization_Test_Product_${YUM_REPOSITORY}" | grep -q "is enabled for this system"
```
Comment (ekohl) [inline bats/fb-content-katello.bats]: Should this be ```${ORGANIZATION}_${PRODUCT}_${YUM_REPOSITORY}```?

File: bats/fb-content-katello.bats [hunk 2]
```diff
@@ -198,8 +199,3 @@ EOF
   skip # because of above
   puppet agent --test && grep -q Lorem /tmp/dummy
 }
-
-@test "Delete an Organization" {
```
Comment (ekohl) [inline bats/fb-content-katello.bats]: Any reason to remove this? Or do you think we should use the other file to do this instead?

File: bats/fb-content-upgrade.bats [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  subscription-manager register --force --org="Test_Organization" --activationkey="Test AK" || true
+  subscription-manager status | grep -q "Current"
+}
+
+@test "check content host is registered" {
+  hammer -u admin -p changeme host info --name $(hostname -f)
+}
+
+@test "enable content view repo" {
+  subscription-manager repos --enable="Test_Organization_Test_Product_${YUM_REPOSITORY}" | grep -q "is enabled for this system"
```
Comment (ekohl) [inline bats/fb-content-upgrade.bats]: Should this be ```${ORGANIZATION}_${PRODUCT}_${YUM_REPOSITORY}```?

File: bats/fb-content-upgrade.bats [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+@test "install katello-agent" {
+  tPackageUpgrade katello-agent && tPackageExists katello-agent
+}
+
+@test "start katello-agent" {
+  service goferd status || service goferd start
```
Comment (ekohl) [inline bats/fb-content-upgrade.bats]: We have a ```tServiceStart``` helper but not one to check if it's running.

File: boxes.yaml.example [hunk 5]
```diff
@@ -35,4 +35,5 @@ centos7-bats-ci:
         - "swapfile.bats"
         - "fb-install-katello.bats"
         - "fb-content-katello.bats"
+	- "fb-destroy-organization.bats"
```
Comment (ekohl) [inline boxes.yaml.example]: The indenting looks weird

File: pipelines/pipeline_katello_31_to_33_upgrade.yml [hunk 6]
```diff
@@ -68,6 +68,18 @@
 - hosts: pipeline-katello-3.1-3.3-centos7
   become: true
   vars:
+    bats_forklift_dir: "/home/vagrant/sync"
```
Comment (ekohl) [inline pipelines/pipeline_katello_31_to_33_upgrade.yml]: The sync directory is non-standard? I'd expect it to be ```/vagrant``` instead.

File: bats/fb-content-katello.bats (hunk 1 above)
Comment (ehelms) [inline bats/fb-content-katello.bats]: Kinda - the label generation makes things a bit harder but I'll add it in.

File: pipelines/pipeline_katello_31_to_33_upgrade.yml (hunk 6 above)
Comment (ehelms) [inline pipelines/pipeline_katello_31_to_33_upgrade.yml]: I would too but lately this is the directory my stuff has been syncing too. I havent looked to see what is causing that though. Will revert for the tim ebeing

File: bats/fb-content-katello.bats (hunk 2 above)
Comment (ehelms) [inline bats/fb-content-katello.bats]: I moved this to another file to facilitate the upgrade testing. If we delete the organization all of the data gets deleted. In an upgrade, having pre-seeded data (in this case by using the content-tests) can illuminate more issues than an empty database due to our data manipulation that occurs during upgrade.

File: bats/fb-content-upgrade.bats (hunk 4 above)
Comment (ehelms) [inline bats/fb-content-upgrade.bats]: I can switch to using that - would prefer to do it in all cases separately though.

**Comment by ehelms** [comment]
Thoughts here @ekohl ?

File: bats/fb-content-upgrade.bats [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+@test "promote content view" {
+  hammer -u admin -p changeme content-view version promote  --organization="${ORGANIZATION}" \
+    --content-view="${CONTENT_VIEW}" --to-lifecycle-environment="${LIFECYCLE_ENVIRONMENT}" --from-lifecycle-environment="Library"
+}
+
+@test "add puppetclass to host" {
+  # FIXME: If katello host is subscribed to itself, should it's puppet env also be updated? #7364
+  # Skipping because of http://projects.theforeman.org/issues/8244
+  skip
+  target_env=$(hammer -u admin -p changeme environment list | grep KT_${ORGANIZATION/ /_}_${LIFECYEL_ENVIRONMENT/ /_}_${CONTENT_VIEW/ /_} | cut -d\| -f1)
```
Comment (ekohl) [inline bats/fb-content-upgrade.bats]: LIFECYEL looks like a typo (even though we skip this code for now).

File: pipelines/pipeline_katello_31_to_33_upgrade.yml [hunk 8]
```diff
@@ -84,6 +95,12 @@
 
 - hosts: pipeline-katello-3.1-3.3-centos7
   become: true
+  vars:
+    bats_forklift_dir: "/home/vagrant/sync"
```
Comment (ekohl) [inline pipelines/pipeline_katello_31_to_33_upgrade.yml]: Did you figure out why this is?

**Comment by ekohl** [comment]
Also when I run this I get:
```
Cannot determine action for task label 'Actions::Katello::Organization::Destroy'/opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.3.1.1/db/migrate/20160923143611_add_action_to_content_view_histories.rb:24:in `block in up'
```

File: pipelines/pipeline_katello_31_to_33_upgrade.yml (hunk 8 above)
Comment (ehelms) [inline pipelines/pipeline_katello_31_to_33_upgrade.yml]: Oops, I did not :/ I will revert for now.

File: pipelines/pipeline_katello_31_to_33_upgrade.yml [hunk 9]
```diff
@@ -84,6 +94,9 @@
 
 - hosts: pipeline-katello-3.1-3.3-centos7
   become: true
+    bats_tests:
```
Comment (evgeni) [inline pipelines/pipeline_katello_31_to_33_upgrade.yml]: this is missing a `vars:` between `bacome` and `bats_tests`

**Comment by ekohl** [comment]
Needs a rebase as well now.

**Pull request closed without merging**